crates/matrix-sdk/src/widget/machine/from_widget.rs

```diff
@@ -245,10 +254,23 @@ impl From<update_delayed_event::unstable::Response> for UpdateDelayedEventRespon
     }
 }
 
-/// The response to the widget that it received the to-device event.
-/// Only used as the response for the successful send case.
-/// FromWidgetErrorResponse will be used otherwise.
-/// This is intentionally an empty tuple struct (not a unit struct), so that it
-/// serializes to `{}` instead of `Null` when returned to the widget as json.
-#[derive(Serialize, Debug)]
-pub(crate) struct SendToDeviceEventResponse {}
+/// Send to device request response.
+#[derive(Serialize, Debug, Default)]
+pub(crate) struct SendToDeviceEventResponse {
+    /// It is possible that sending to some device failed (failed to encrypt or
+    /// failed to send)
```

Comment:
Ok, it's possible that sending may fail, but could we explain how this field relates to this fact? What's contained in the map? What does a `None` mean?

Can't we fold `None` into the empty `BTreeMap` case and skip the serialization if the map is empty?

Reply:
You can implement arbitrary functions that tell you when serialization should be skipped. Most `is_empty` functions for any struct should work as well.